### DISCUSSION TITLE
Timing for 3A+ on an NTSC C128?

### DISCUSSION THREAD

**Discussion by doughamm** [frntc/RAD#2] [Q&A]
Hey there!  I'm having trouble with the menu screen giving graphical glitches on an NTSC C128.  Any suggestions for timing or other configuration settings?  TIA!

**Comment by frntc** [comment]
If it's only graphical glitches in the menu (and the rest is working) it's probably one of these two timings (see rad.cfg):
WAIT_ENABLE_RW_ADDRLATCH
WAIT_ENABLE_DATA_WRITING

For C128 also the WAIT_OFFSET_CBTD-parameter is relevant, but most likely not the problem.

Try modifying these value (+/- 10 steps at first).

**Reply by paulhocker** [reply]
What settings worked for you?

**Reply by frntc** [reply]
nik99999-s has been blocked, because of posting ads in the discussion

**Comment by paulhocker** [comment]
I have 2 C128 where stock settings produce little to no glitches, while 3 others have horrible glitches and garbled screens. I will try some of the settings above to see if they help.

I am using 3B+ -- I think I have a 3A+ I can try with as well.

**Reply by doughamm** [reply]
Interesting!  My 128DCR has no glitches, and it happens to be where I use the RAD regularly, so I've not been impacted by the glitches on my two flatties.  If you happen to find settings that don't glitch out, I'd love to know what they are.

**Comment by paulhocker** [comment]
One thing I learned today is that putting a comment at the end of  a setting removes that setting entirely. For example

```
WAIT_ENABLE_RW_ADDRLATCH    475 # 467
```
completely breaks the unit and nothing works

**Comment by frntc** [comment]
you might want to try the new firmware. you can then adjust/try out timings from within the menu (and save to SD)

**Reply by paulhocker** [reply]
Wow. I actually just received the pcb on Thursday and I must have downloaded the old version. At any rate, it still looks like I still have the same glitches out of the box. I will try messing around with the settings some more. Thanks. Does the firmware mess around with the IEC at all? After pressing X to return to the C128 my SD2IEC tries to boot a file, and get a file not found error. Doing a Run/Restore seems to correct the issue, but it is just odd. Without the Cartridge in, it starts up normally. Ill try to post a video to show you.

**Comment by genetoye** [comment]
Anybody found any good settings?  I'm seeing the same garbled RAD screen on my C128.  Same RAD works great on a breadbin with defaults.

**Comment by paulhocker** [comment]
Not yet. I played around with some of them and did not seem to make a difference. I also just received a 3A+ and tried that but with the same results. It seems that once you leave the title screen it is fine, except for the problems I mentioned above the my sd2iec device locking up

**Comment by paulhocker** [comment]
I still need to pull out my 128D to see if it has the same issue. So far three flats have the problem. It seems like a VIC timing thing, but I am probably wrong about that.

**Comment by frntc** [comment]
I'd need some more information about your C128s (board revision, VIC, possible internal upgrades/modifications) to reproduce -- in case I can (e.g. NTSC C128 are close to impossible to get here)

**Reply by genetoye** [reply]
I have two C128s.  One is stock and the other has JiffyDos and 64kVDC upgrades.  Both are the 310831 R7 motherboard with 8564R5C VIC chips.  Both exibit the corrupted RAD screen and fail when the eXit command is used. The 128 colors are displayed on the monitor but there is changing garbage and no BASIC header or prompt.
![PXL_20230317_225632389](https://user-images.githubusercontent.com/31168026/226066334-5472bc38-df25-47e0-8392-0ebf169611d9.jpg)
![PXL_20230317_225639420](https://user-images.githubusercontent.com/31168026/226066362-da425500-1ddc-4949-b89d-1fb70b75029b.jpg)

**Reply by genetoye** [reply]
I am using this board with a 3A+.  https://www.ebay.com/itm/165957565785

**Reply by genetoye** [reply]
FYI, the HCT30 at the top of the board is not installed.  Is this an issue?  It works fine with an NTSC C64.

**Reply by genetoye** [reply]
Sorry, the 310381 R7 board.

**Reply by frntc** [reply]
sorry, overlooked this thread completely. could you solve your problem? missing HCT30 is intended.

**Comment by frntc** [comment]
I'll see how much of this setup I can reproduce.  Did you build the RAD yourself? (can you check the soldering of the CBTD3861, in particular pins 6 and 18? <- asking that because exit also doesn't work, because actually RAD doesn't do anything when you exit :-) )

**Reply by genetoye** [reply]
I purchased an assembled unit from ebay (see link above). Soldering job is very professional.

**Comment by paulhocker** [comment]
FWIW -- I just received my SideKick64 today and it works flawlessly on my C128. I hope to have some time this weekend to do some more troubleshooting on the RAD and get your some model #'s to help.

**Comment by paulhocker** [comment]
My 128D show no signs of graphical glitches with or without the expansion board. I will open it up later this weekend to see what revision the board is. The RAD and SIDEKICK are so cool.

**Comment by doughamm** [comment]
Hey folks -- circling back with my own experiences.  This is with a 3A+ RAD:

- NTSC and PAL C64's in my collection: RAD works very well on default settings and is highly tolerant of tinkering.  REU test program passes.  This is bus (3rd party C64 supply!) or external Pi powered.
- NTSC C128DCR: RAD works very well on default settings, even with X-pander 3 in the mix, and is highly tolerant of tinkering.  REU test program passes.  Bus or external Pi powered.
- 2 different NTSC C128 flats:
     - Now, I found this interesting: on the flat 128, I found that if bus powered the Pi would sometimes not boot fully and the red/green lights would actually go off after a few seconds.  I checked the voltages on my power supply and they're ample (5.3V / 9.8V before load).  So these are all tested with an external CanaKit high-quality Pi power supply.
     - At default settings there is visual artefacting and the REU test program errors out saying the REU is broken.  This also appears to hang the Pi since pressing the Menu button just gives a green or red screen until the Pi is power cycled.
     - Setting Enable Data to mid-high 700's (I know, radical increase) reduced the visual artefacts but the REU test still fails
     - Once in a blue moon I can see a single successful block read in the REU test but it still always fails.

So there's definitely something special with the NTSC flatties.  It may be noteworthy that when it came to the C128 Evie Salomon had no end of difficulty getting the BackBit Pro working at 2MHz (though 1MHz was fine) -- she eventually went so far as to make her own PLA replacement with a compatibility mode that fixed the 2MHz issue.  I have one and tested the RAD with it for fun, but it didn't help (artefecting was worse, suggesting the different PLA requires significantly different RAD settings).  This is all to say, I acknowledge how difficult this is to nail down on such an idiosyncratic computer as the first C128.  :)

Just thinking out loud here...anyone have a hotline to Bil Herd?  He and I have chatted back and forth in the past related to the BackBit issue; he may be able to shed light on what's going on here.

**Reply by RetroBard** [reply]
I have a flat PAL C128 with JiffyDos myself and am experiencing graphical artefacts on the RAD menu. I’ve tried to tinker with the timings to no awail. Probably haven’t found the correct settings yet. 

I’ve also been trying out Sonic the Hedgehog. With that, I can’t seem to be able to get all the way into the gameplay. The best I’ve achieved so far is a blank screen and the music of the first level. 

Granted, I haven’t yet tried with a C64. I’ve got one, though. I also have a C128D and a C64C. All are PAL units.

**Reply by RetroBard** [reply]
I dug out my breadbin. The menu is working great in it and Sonic works and plays too.

**Comment by frntc** [comment]
I'll prepare an update in the next days. As I don't have NTSC C128Ds I can't really test that (and especially the 128D are a bit deviating regarding bus timings), but the changes allowed me to get it run on a C64 someone sent to me that caused problems previously.

**Comment by futureretrofusion** [comment]
Hi Frntc/Rad

Just wanted to let you know I've started making both Sidekicks and Rad
cards here in Australia and have had lots sales. Hopefully this will lead
to lots attention given to your amazing projects.

One question how hard is it to have the RAD ready .d64 files?

Cheers and let me know if there is anything I can do to contribute.

Cheers,

Ian

**Reply by frntc** [reply]
Hi Ian,

can you contact me via mail?

About the D64-support: if it's about starting a PRG from a D64-image -- this would be diligence work only. If you're talking about disk emulation (even via kernal vectors) that's a different story as the RAD is not meant for this. It would require the same approach as the Turbo Chameleon (emulating the CPU plus everything that you want)

**Comment by RetroBard** [comment]
Hi Frntc,

Any update on the new firmware?

**Reply by frntc** [reply]
I've been quite busy in the last weeks (and digesting some annoying things about licenses) and so this new firmware is a lonely existence on my hard drive :-) I have to put together a NTSC C128 to test for timing values, then I'll assemble a new release.

**Reply by doughamm** [reply]
Rad!  :)  Can't wait.  If you want beta testing feel free to reach out.

**Reply by RetroBard** [reply]
Awesome! Looking forward to this! My C128 is a PAL unit, but can try it out too.

**Comment by CBMfan** [comment]
I have managed to put together RAD with Pi 3 Model B v1.2 and it does boot but garbled. It is a CBM128 NTSC and I see others had same issues. I want to try to work on it and fix it but I am not sure where to start. I don't want to mess it up more than it is already.
I see up there a suggestion of the 3 parameters. Any other ones that affect NTSC flat 128?

**Reply by frntc** [reply]
is it a Pi Model B (not B+) with a BCM2837 SoC? (should be printed on the large IC) In this case you have a SoC which is older than the supported 3A+ and 3B+ models and you can be lucky with overclocking (values in config.txt), but there's a good chance that you're not.

If it's a BCM2837B0 SoC you should experiment with the 3 last timing values. The last one typically can vary between 40 and 63 (or so), the two before can be quite different, but should have a difference of 150-220. Hope that helps!

**Comment by insanity213** [comment]
Did anyone ever get their C128 working with the RAD?  Another flattie here with the same problems.

**Reply by RetroBard** [reply]
There’s a newer firmware available, but I haven’t tried that with my C128.

**Reply by insanity213** [reply]
I haven't had any luck with the 0.12 firmware.

**Reply by frntc** [reply]
can you describe the problems/artifacts? did you power the RAD externally? are you using a Zero2/3A+/3B+ or a Pi3 without "+"?

**Reply by insanity213** [reply]
The artifacts looks like random glitches in the character rom for lack of a better description.  Random flashes of junk on the screen here and there.  Had similar visual artifacts on various C64 motherboard models I've used this device with but was able to find timings that cured the artifacts and let Sonic fully load to REU and work on all of those machines.  Powering internally or externally didn't seem to have any impact.  The PSU I'm using is capable of 5a on the 5v rail.  I went as far as adding a 1000 uf cap to pins 1 and 3 on the cart slot just in case inrush current was an issue.  Same issues persist with a Pi 3b+ or a Zero 2w.

**Reply by insanity213** [reply]
I should probably also mention this is a Neo128 v4 motherboard with Saruman SRAM.  I do have my original CBM manufactured board here with DRAM on it but IIRC it did the same thing.  I can break it out tomorrow and test though.

**Comment by metaone70** [comment]
I have a flat C128 and have the same type of glitches. The music sounds ok but I could not get rid of the glitches with different settings. My PAL C128 has 310381 Rev 7 motherboard, 8566R3 VIC-II chip and I'm using my RAD with Rbpi Zero 2W. I'll be checking this thread for further developments.

**Reply by insanity213** [reply]
I've given up trying to find good timings.  The cart works on one of my 64's but doesn't like either of my 128s or my Sixtyclone.

**Reply by Nivlekk** [reply]
I'm also experiencing the same problem with my C128 flat PAL, assy 310381, rev 7.
The menu shows glitches but the menu music plays fine. When booting the system in either C128 or C64 mode, random characters appear all over the screen and the computer becomes unusable as soon as the RAM configuration message appears at the bottom of the screen.
My RAD works perfectly fine on my C64 breadbin PAL.
I've spend quite some time messing with all the different timings but without any luck.

**Reply by insanity213** [reply]
FWIW, I found a 60pf variable cap in my parts bin this morning so I added it between the phi2 input and gnd on the RAD cart.  I was able to get the RAD menu to be video glitch free by tweaking the cap but I crash shortly after hitting the start screen on sonic.  Something like this might work better after the bus transceiver instead of before it?  Idk, just thought I'd throw it out there.

**Comment by paulhocker** [comment]
Here are a couple of links to videos of some sessions I did this morning with my NTSC flat. I had not tried testing this cart in a while so I made sure to use the latest release. Overall the experience was much better with occasional artifacting, and much better actually dropping to the OS with the REU activated. In the previous release it would rarely drop to OS without crashing. 

I do plan on testing another couple of flats I have to see how they do as well.

https://youtu.be/yZOqC4_Qyzw

https://youtu.be/XmoS5G_Kpxc

**Reply by frntc** [reply]
How do you get this image quality out of a C128? (some video output modification? does it behave differently without?)

This number of artifacts in the menu shows some severe problems, I don't think it makes sense trying anything unless you get rid of those. Sorry for maybe asking again, you did experiment with the timing settings on this one already?

**Reply by paulhocker** [reply]
Using this

https://www.tindie.com/products/c0pperdragon/viciidizer128/

I do not think it is the dizer because I had them before without it -- i wanted to use OBS record it for you so that you can see the results.

I have experimented with the timing before, but not since before the latest release so I will try again.

It is interesting to me that sometimes it works with zero issues, and then a couple resets later it is all weird. And if I recall, PAL users are not experiencing this as much?

It is actually working better than it did before. :)

**Reply by insanity213** [reply]
FWIW, basically same results here both on the Neo128 w/ c0pperdragon component mod and an all original CBM board (both NTSC).  I've got a PAL board here but missing it's keyboard connector.  If I can rig something up for a keyboard I'll try the RAD with it at some point.
